File: YYReader/Parsing/LoadedHTML.swift
import Foundation

struct LoadedHTML: Sendable {
    enum RetrievalKind: String, Sendable {
        case urlSession
        case webKit
    }

    let requestedURL: URL
    let finalURL: URL
    let html: String
    let retrievalKind: RetrievalKind
}

File: windows/YYReader.Windows.Core/Parsing/LoadedHtml.cs
namespace YYReader.Windows.Core.Parsing;

public sealed record LoadedHtml(
    Uri RequestedUri,
    Uri FinalUri,
    string Html,
    HtmlRetrievalKind RetrievalKind);

public enum HtmlRetrievalKind
{
    UrlSession,
    WebView2
}
